test/fixture.py

```diff
@@ -51,6 +51,46 @@ def open_mock(content=None, **kwargs):
         rv.__iter__ = lambda x: iter(stream.readlines())
         yield rv
 
+@contextmanager
+def open_mock_many(file_content_map=None, **kwargs):
+    """
+    Mock out access to one or many files opened using the builtin "open".
+    :param file_content_map: A dictionary of path : file_contents
+    :type file_content_map: dict[str,str]
+    :param kwargs:
+    :return:
+    """
+    file_content_map = file_content_map or {}
+    for key, value in file_content_map.items():
+        file_content_map[key] = (mock_open(read_data=value), value, six.StringIO())
+
+    def get_file(path, *args, **kwargs):
+        """
+        The side effect that will allow us to "open" the right "file".
+        Not for use outside open_mock_many.
+        :param path: The path which is passed in to the built
+        :param args:
+        :param kwargs:
+        :return:
+        """
+        try:
+            rv, file_contents, content_out = file_content_map[path]
+        except KeyError:
+            if six.PY2:
+                raise IOError(2, 'No such file or directory')
+            else:
+                raise OSError(2, 'No Such file or directory')
+
+        stream = six.StringIO(file_contents)
+        rv = rv.return_value
+        rv.write = lambda x: content_out.write(x)
+        rv.content_out = lambda: content_out.getvalue()
+        rv.__iter__ = lambda x: iter(stream.readlines())
+        return rv
```

Comment:
Instead of all this, I think you can probably just do something using [`mock_open`](https://docs.python.org/3/library/unittest.mock.html#mock-open)?

Reply:
It does, but mock_open does not track what you write.

Reply:
Of course for the tests this is used in the "content_out" is not used. Should I still remove it?

Reply:
I wouldn't sweat it then. Since we are using similar code elsewhere. Could be worth a tech-debt backlog item to clean up a bit later though.